package.json

```
@@ -1,150 +1,16 @@
-    "ember-render-helpers": "^0.2.0",
-    "ember-style-modifier": "^0.8.0 || ^1.0.0 || ^2.0.0 || ^3.0.0",
-    "findup-sync": "^5.0.0",
-    "fs-extra": "^11.0.0",
-    "resolve": "^1.18.1",
-    "silent-error": "^1.0.1",
-    "tracked-toolbox": "^1.2.3 || ^2.0.0"
+    "prepare": "husky install",
+    "test": "pnpm --filter 'test-app' test",
+    "test:ember": "pnpm --filter 'test-app' test:ember"
   },
   "devDependencies": {
-    "@babel/eslint-parser": "7.23.3",
-    "@babel/plugin-proposal-decorators": "7.23.7",
-    "@ember/optional-features": "2.0.0",
-    "@ember/string": "3.1.1",
-    "@ember/test-helpers": "3.2.1",
-    "@embroider/test-setup": "2.1.1",
-    "@glimmer/component": "1.1.2",
-    "@glimmer/tracking": "1.1.2",
-    "@release-it-plugins/lerna-changelog": "6.0.0",
-    "bootstrap": "5.3.2",
-    "broccoli-asset-rev": "3.0.0",
-    "chai": "4.4.1",
-    "chai-things": "0.2.0",
-    "concurrently": "8.2.2",
-    "ember-a11y-testing": "6.1.1",
-    "ember-cli": "5.5.0",
-    "ember-cli-app-version": "6.0.1",
-    "ember-cli-blueprint-test-helpers": "0.19.2",
-    "ember-cli-content-security-policy": "2.0.3",
-    "ember-cli-dependency-checker": "3.3.2",
-    "ember-cli-fastboot": "4.1.2",
-    "ember-cli-fastboot-testing": "0.6.1",
-    "ember-cli-inject-live-reload": "2.1.0",
-    "ember-cli-sri": "2.1.1",
-    "ember-cli-terser": "4.0.2",
-    "ember-cli-test-loader": "3.1.0",
-    "ember-cli-yuidoc": "0.9.1",
-    "ember-compatibility-helpers": "1.2.7",
-    "ember-load-initializers": "2.1.2",
-    "ember-page-title": "8.2.0",
-    "ember-qunit": "7.0.0",
-    "ember-resolver": "11.0.1",
-    "ember-sinon": "5.0.0",
-    "ember-sinon-qunit": "7.4.0",
-    "ember-source": "5.5.0",
-    "ember-source-channel-url": "3.0.0",
-    "ember-template-lint": "5.13.0",
-    "ember-template-lint-plugin-prettier": "5.0.0",
-    "ember-try": "3.0.0",
-    "eslint": "8.56.0",
-    "eslint-config-prettier": "9.1.0",
-    "eslint-plugin-ember": "11.12.0",
-    "eslint-plugin-hbs": "1.0.3",
-    "eslint-plugin-n": "16.6.2",
-    "eslint-plugin-prettier": "5.1.3",
-    "eslint-plugin-qunit": "8.0.1",
-    "execa": "5.1.1",
-    "glob": "10.3.10",
-    "gulp": "4.0.2",
-    "gulp-connect": "5.7.0",
-    "gulp-insert": "0.5.0",
-    "gulp-rename": "2.0.0",
-    "gulp-transform": "3.0.5",
     "husky": "8.0.3",
-    "lint-staged": "15.2.0",
-    "loader.js": "4.7.0",
-    "merge-stream": "2.0.0",
-    "mocha": "10.2.0",
-    "prettier": "3.0.3",
-    "qunit": "2.20.0",
-    "qunit-dom": "3.0.0",
-    "release-it": "16.3.0",
-    "sinon": "17.0.1",
-    "striptags": "3.2.0",
-    "webpack": "5.89.0"
-  },
-  "peerDependencies": {
-    "@glimmer/component": "^1.0.4",
-    "@glimmer/tracking": "^1.0.4",
-    "ember-source": ">=4.8.0"
-  },
-  "engines": {
-    "node": "18.* || >= 20"
+    "lint-staged": "15.2.0"
   },
-  "publishConfig": {
-    "registry": "https://registry.npmjs.org"
+  "lint-staged": {
+    "*.js": "eslint --cache --fix"
```

I think we should fix all fixable linting issues before committing. At least we should run Prettier for Glimmer templates as well. But that's independent from this PR. I only noticed when reviewing. And wanted to keep a reminder.

---

That indeed sounds interesting. Does lint-staged also support modifying staged files?

---

I'm also not that experienced with [lint-staged](https://github.com/lint-staged/lint-staged) to be honest.

I think we are capturing majority of the files being changed in regular commits with our existing linters. So maybe we just want to run linting _always_ regardless and skip filtering on specific files?

I wonder if something like this would work:

```suggestion
    "*": "pnpm run lint:fix"
```

We should quickly check how long it takes. For me lint-staged and similar pre-commit hooks are more annoying than helpful. I have configured my editor to lint on save anyways. But if I recall correctly we primarily added it to reduce barrier for new contributors.

---

The fixing works, but after testing, it seems that using `"*"` as lint-staged target did not work as expected.

I've changed the configuration now so it will run eslint and ember-template-lint. We should include Typescript when we have that as well.

I think that further modifications to the lint-staged configuration should go in a separate PR, the monorepo migration is already difficult to oversee 😅